addons/crm/models/crm_stage.py

```diff
@@ -22,31 +22,19 @@ class CrmStage(models.Model):
     _rec_name = 'name'
     _order = "sequence, name, id"
 
-    @api.model
-    def default_get(self, fields):
-        """ As we have lots of default_team_id in context used to filter out
-        leads and opportunities, we pop this key from default of stage creation.
-        Otherwise stage will be created for a given team only which is not the
-        standard behavior of stages. """
-        if 'default_team_id' in self.env.context:
-            ctx = dict(self.env.context)
-            ctx.pop('default_team_id')
-            self = self.with_context(ctx)
-        return super().default_get(fields)
-
     name = fields.Char('Stage Name', required=True, translate=True)
     sequence = fields.Integer('Sequence', default=1, help="Used to order stages. Lower is better.")
     is_won = fields.Boolean('Is Won Stage?')
     requirements = fields.Text('Requirements', help="Enter here the internal requirements for this stage (ex: Offer sent to customer). It will appear as a tooltip over the stage's name.")
-    team_id = fields.Many2one('crm.team', string='Sales Team', ondelete="set null",
-        help='Specific team that uses this stage. Other teams will not be able to see or use this stage.')
+    team_ids = fields.Many2many('crm.team', relation='crm_stage_crm_team', column1='crm_stage_id', column2='crm_team_id',
```

Comment:
Why not use default values?
```suggestion
    team_ids = fields.Many2many('crm.team',
```

Reply:
But then upgrade script is not correct anymore, table name is not correct.